infra/test/integration/simple_example/simple_example_test.go

```diff
@@ -96,6 +111,85 @@ func TestSimpleExample(t *testing.T) {
 			return false, nil
 		}
 		utils.Poll(t, isServing, 20, time.Second * 3)
+
+		// Check Cloud Function source code exists
+		annotateHttpFunction := gcloud.Run(t, fmt.Sprintf("functions describe %s --region %s --gen2 --format=json", annotateHttpFunctionName, annotateHttpFunctionRegion), gcloudArgs)
```

Comment:
issue: this looks like there are multiple test cases put into a single test method, `TestSimpleExample`. Break different test cases into separate test methods.

Reply:
If we separate them into different test case methods, the code indeed looks better. However, there are two main reasons for keeping them in a single test function:

1. The function name in this context is also related to the step format in build/int.cloudbuild.yaml. If we separate the test cases into multiple test methods, we would need to add a verify step to each of the method in int.cloudbuild.yaml to execute these tests. While it has been tested and proven to work, it is unclear whether executing only the verify stage for a test case method aligns with the conventions of CFT testing. This is because a complete CFT test cycle consists of the init, apply, verify, and teardown stages. However, during the execution of the apply and teardown stages, there are lengthy operations performed by Terraform on the Cloud resources.

2. In the subsequent test cases, we need some parameters that require more complex logic to obtain. These parameters can be obtained during the preceding test cases. Therefore, if we separate them, it would be less efficient in terms of performance.

Reply:
Okay, that's fair for now. If multiple test cases isn't feasible, then we can live with a single jumbo test. Thanks!

Reply:
fixed in [06ee2da](https://github.com/GoogleCloudPlatform/terraform-ml-image-annotation-gcf/pull/28/commits/06ee2da76f45b38c4003a3f8c7ff127ac966b143)